app/models/git_repository.rb

```diff
@@ -91,6 +91,8 @@ def update_repo
       _log.info(message)
       worktree.send(:pull)
       _log.info("#{message}...Complete")
+    rescue MiqException::MiqUnreachableError => err
+      raise "Failed to connect with [#{err.class}: #{err}]"
```

Comment:
Is there a reason this is being rescued here instead of in [`ManageIQ::Providers::AnsibleRunnerWorkflow#checkout_git_repository`](https://github.com/ManageIQ/manageiq/blob/cf751304/app/models/manageiq/providers/ansible_runner_workflow.rb#L178) ?

Also, is it possible to raise a different error instead that automate would then pickup on and avoid running `check_complete` a 100 times?  @tinaafitz ?

Reply:
yeah, so for the other prov state machines, the task update system prevents (in theory) this kind of thing from happening. and i don't mean to point fingers ... but [the pr in which this particular state machine got introduced](https://github.com/ManageIQ/manageiq/pull/17759) makes a point of _not_ using that mechanism, if you read the comments Madhu in particular had about it. it's not so much that there's a better error to raise here, it's that this beast is just different enough that i expect it'll need to be handled somewhat differently.

Reply:
For my sake, can you link to where in the code this might happen? (in the "task update system" that isn't being used by this mechanism)

Reply:
sure, sorry. the handling i'm talking about is like the stuff in https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Service/Generic/StateMachines/GenericLifecycle.class/__methods__/execute.rb#L19. that logic's missing here: we're only checking the status of the job but the associated task errors are ignored.